### PR TITLE
[Concept Entry] Sklearn linear-regression-analysis

### PR DESCRIPTION
**Pull request by SaviDahegaonkar**
### Description

Created a new entry on the Linear Regression Analysis concept under Sklearn.

### Issue Solved
#4823 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/sklearn/concepts/linear-regression-analysis/linear-regression-analysis.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+---
+Title: 'Linear Regresssion Analysis'
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/linear-regression-analysis/linear-regression-analysis.md:2]: There's an extra 's' in Regression:

```suggestion
Title: 'Linear Regression Analysis'
```

File: content/sklearn/concepts/linear-regression-analysis/linear-regression-analysis.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+Description: 'A machine learning technique that involves training a model to predict a dependent variable from one or more independent variables, assuming a linear relationship between them.'
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/linear-regression-analysis/linear-regression-analysis.md:3]: We can simplify this as follows:

```suggestion
Description: 'A machine learning technique used to predict a dependent variable from one or more independent variables, assuming a linear relationship between them.'
```

File: content/sklearn/concepts/linear-regression-analysis/linear-regression-analysis.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  - 'Linear Regression'
+  - 'Machine Learning'
+  - 'Scikit-learn'
+  - 'Supervised Learning'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/machine-learning'
+---
+
+In sklearn,**Linear Regression Analysis** is a technique for simulating the relationship between a dependent variable and one or more independent variables. In simpler terms, it can use it to predict the value of one variable based on the value of another, assuming a linear relationship between them.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/linear-regression-analysis/linear-regression-analysis.md:17]: There's a missing space in the first line, and also we can skip the second sentence as it is just repitition.

```suggestion
In sklearn, **Linear Regression Analysis** is a machine learning technique used to predict a dependent variable based on one or more independent variables, assuming a linear relationship.
```

File: content/sklearn/concepts/linear-regression-analysis/linear-regression-analysis.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+In simple linear regression, we predict the dependent variable `Y` using a single independent variable `X`, fitting the data to a straight line, often called as the `regression line`. The equation for the line is:
+
+$$
+Y = \beta_0 + \beta_1 X + \epsilon
+$$
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/linear-regression-analysis/linear-regression-analysis.md:23]: Please explain each parameter with some context.

File: content/sklearn/concepts/linear-regression-analysis/linear-regression-analysis.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+print("Coefficient (β₁):", model.coef_[0])
+print("Predictions:", predictions)
+plt.scatter(X, y, color='red', label='Data Points')
+plt.plot(X, predictions, color='green', label='Regression Line')
+plt.xlabel('X')
+plt.ylabel('y')
+plt.title('Simple Linear Regression')
+plt.legend()
+plt.show()
+
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/linear-regression-analysis/linear-regression-analysis.md:122]: It is missing import for `plt`.

```suggestion
import numpy as np
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression

# Example data: Hours studied (X) and scores (y)
X = np.array([[1], [2], [3], [4], [5]])  # Hours studied
y = np.array([2, 4, 5, 4, 5])  # Scores achieved

# Create a linear regression model
model = LinearRegression()

# Train the model on the data
model.fit(X, y)

# Get the model's predictions
predictions = model.predict(X)

# Output the results
print("Intercept (β₀):", model.intercept_)
print("Coefficient (β₁):", model.coef_[0])
print("Predictions:", predictions)
plt.scatter(X, y, color='red', label='Data Points')
plt.plot(X, predictions, color='green', label='Regression Line')
plt.xlabel('X')
plt.ylabel('y')
plt.title('Simple Linear Regression')
plt.legend()
plt.show()

```

File: content/sklearn/concepts/linear-regression-analysis/linear-regression-analysis.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+```
+
+The code outputs the following result:
+
+```shell
+Intercept (β₀): 0.0
+Coefficient (β₁): 1.0
+Predictions: [1. 2. 3. 4. 5.]
+```
+![Simple Linear Regression]
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/linear-regression-analysis/linear-regression-analysis.md:88]: The image description should be more descriptive.

```suggestion
![A scatter plot showing the data points (red dots) and the corresponding regression line (green) representing the simple linear regression model.]
```

**Review by PragatiVerma18 — CHANGES_REQUESTED** [review]
Hey @SaviDahegaonkar, thanks for adding this entry. I have added some comments, please check and do the required changes.

**Comment by SaviDahegaonkar** [comment]
> @SaviDahegaonkar just a suggestion i think you need to Run `yarn format:verify` in order to get rid of failing checks. Also for lint run `yarn lint:md`

Yup, Will keep this in mind. Thanks @Anshgrover23

**Comment by SaviDahegaonkar** [comment]
Hey @PragatiVerma18, I made the changes you suggested. Please have a look at them.

**Review by PragatiVerma18 — APPROVED** [review]
Hey @SaviDahegaonkar, thanks for doing the changes, this looks good to go!
